### ISSUE TITLE
Models (ORM): Plural class name "Invoices" used for single invoice

### ISSUE THREAD

**Issue by Chealer** [phalcon/docs#2606]
[The Models page](https://docs.phalcon.io/4.0/en/db-models#overview) defines an _Invoices_ class:
```
<?php

namespace MyApp\Models;

use Phalcon\Mvc\Model;

class Invoices extends Model
{

}
```

```
<?php

use MyApp\Models\Invoices;

$invoice = new Invoices();

$invoice->inv_cst_id      = 1;
$invoice->inv_status_flag = 1;
$invoice->inv_title       = 'Invoice for ACME Inc.';
$invoice->inv_total       = 100;
$invoice->inv_created_at  = '2019-12-25 01:02:03';

$result = $invoice->save();

if (false === $result) {
    
    echo 'Error saving Invoice: ';

    $messages = $invoice->getMessages();

    foreach ($messages as $message) {
        echo $message . PHP_EOL;
    }
} else {

    echo 'Record Saved';

}

```

The class name is plural (_Invoice**s**_), but the second code fragment shows an instance of the Invoices class represents a single invoice.

I imagine the class was named "Invoices" because it corresponds to a database table named _Invoices_, but this could confuse readers and does not provide a good example, since class names which represent a single object are singular in object-oriented languages.

I recommend to either:

1. rename the class Invoice and use setSource() to map it to a differently-named table
2. or, failing that, add PHPDoc to Invoices indicating what it represents (which would not be a bad thing anyway)

**Comment by niden** [comment]
This is a well known issue for pretty much every framework or PHP application. How do you name your models. 

It is completely arbitrary - in the examples we used the plural `Invoices`. So this brings the paradox you mentioned:

```php
$invoice = new Invoices();
```

If we reverse this and call it `Invoice` then we will have this:

```php
$invoice = new Invoice();
```

but what would happen if we get many results back from the database?

```php
$invoices = Invoice::find(
    [
        'conditions' => 'inv_total > :total:'
        'bind'       => [
            'total' => 100,
        ]
    ]
```
In the above the `$invoices` variable represents a collection but the model itself uses the singular word.

There is no real way to deal with this that would satisfy both. Either you have `Invoice` or `Invoices`. One could potentially create two models and use them accordingly but that is going to be a waste of resources.

**Comment by Chealer** [comment]
I agree that for a class which would only provide methods dealing with a variable number of instances, plural would make more sense. Java uses the plural in standard libraries, but only for classes which cannot be instantiated.

That being said, it **is** a convention to use singular for class names (except when the plural makes more sense), so there is a user expectation that a plural name denotes something particular (such as a class which cannot be instantiated). But I am fine with the second solution I offered if Phalcon wants to stick to plural.

**Comment by niden** [comment]
> or, failing that, add PHPDoc to Invoices indicating what it represents (which would not be a bad thing anyway)
@Chealer what do you mean by that? In the docs add a docblock?

**Comment by Chealer** [comment]
Yes @niden :
```
<?php

namespace MyApp\Models;

use Phalcon\Mvc\Model;

/**
 * An invoice
 */
class Invoices extends Model
{

}
```

**Comment by ruudboon** [comment]
@Chealer Would you mind sending in a pull request for this? https://github.com/phalcon/docs/blob/4.0/en/db-models.md

**Comment by Chealer** [comment]
@ruudboon as implementation is trivial and there are more than 1 solution to this problem, I would rather leave a regular Phalcon developer decide which is best.

**Comment by ruudboon** [comment]
@ruudboon I was referring to the Docblock example in the documentation.
